src/mono/mono/mini/CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ if(HOST_WIN32)
 endif()
 
 # ICU
-if(HAVE_SYS_ICU)
+if(HAVE_SYS_ICU AND !HOST_WASI)
   if(STATIC_ICU)
```

Comment:
There are failures on CI wrt to linking icu, and this line might be the cause.

Reply:
Well spotted - thanks!